### PR TITLE
fix missing dropout on the embeddings

### PR DESCRIPTION
**Pull request by JulesGM**
There should be dropout on the embeddings, before feeding them to the encoder layers. See the reference implementation:
https://github.com/tensorflow/models/blob/master/official/transformer/model/transformer.py#L124
and see its absence here:
https://github.com/jadore801120/attention-is-all-you-need-pytorch/blob/master/transformer/Models.py#L87

see https://github.com/jadore801120/attention-is-all-you-need-pytorch/issues/69

### CODE REVIEW

**Comment by jadore801120** [comment]
@JulesGM 
It is my mistake to check pull request before I modify a lot on my code. 😥
Thanks for your contribution!
All the best, 
Yu-Hsiang
